niri-ipc/src/lib.rs

```diff
@@ -628,3 +632,14 @@ impl FromStr for ScaleToSet {
         Ok(Self::Specific(scale))
     }
 }
+
+/// Display mapped workspace information
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct IpcWorkspace {
+    /// Id of the workspace
+    pub id: u32,
+    /// Monitor id the workspace is on
+    pub monitor: usize,
+    /// If the workspace is currently active
+    pub active: bool,
```

Comment:
So, a couple of thoughts.

First, let's rename `id` to `idx`. So that we can expose `id` later when we have a more complete picture of the IPC.

Second, we don't actually have any stable output mapping either (`niri msg outputs` returns a `HashMap`). Let's use the output name. Plus, let's make it an `Option` so it works for the "no connected monitors" case too.

Third, let's add a newly-added workspace `name`.

```suggestion
    /// Index of the workspace on its monitor.
    ///
    /// This is the same index you can use for requests like `niri msg action focus-workspace`.
    pub idx: u8,
    /// Optional name of the workspace.
    pub name: Option<String>,
    /// Name of the output that the workspace is on.
    ///
    /// Can be `None` if no outputs are currently connected.
    pub output: Option<String>,
    /// Whether the workspace is currently active on its output.
    pub is_active: bool,
```